src/Ushahidi/Modules/V5/Repository/Role/RoleRepository.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Ushahidi\Modules\V5\Repository\Role;
+
+use Ushahidi\Modules\V5\Models\Role;
+use Illuminate\Pagination\LengthAwarePaginator;
+
+interface RoleRepository
+{
+ /**
+     * This method will fetch all the Role for the logged user from the database utilising
+     * Laravel Eloquent ORM and return them as an array
+     * @param int $limit
+     * @param int $skip
+     * @param string $sortBy
+     * @param string $order
+     * @return Role[]
+     */
+    public function fetch(
+        int $limit,
+        int $skip,
+        string $sortBy,
+        string $order,
+        array $search_data
+    ): LengthAwarePaginator;
```

Comment:
We probably should return Collection and convert it into LengthAwarePaginator in the Controller. If we were to use that in CLI we'd have no use in Paginator there.

Reply:
the collection doesn't return the paging info (total and links ) . without these ingo we can't create the pagination 
any suggestions?

Reply:
We can skip it for now. I thought we can create it from Collection but it seems to be a bit more complex than that.